packages/cli/src/error_handler.ts

```diff
@@ -92,12 +107,16 @@ const handleError = (
     if (errorHasCauseMessage(error)) {
       printer.print(`Cause: ${error.cause.message}`);
     }
+    await usageDataEmitter?.emitFailure(error, { command: command ?? '' });
```

Comment:
should the default be empty string or undefined? Empty string could be ambiguous in cases like `amplify --version` or `amplify --help`

Reply:
This should not emit metrics for help and version commands. And the dimension shape expects string, it won't be happy with `undefined`